node/lib/stat_emitter.js

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var inherits = require('util').inherits;
+var EventEmitter = require('./event_emitter');
+var Stat = require('./stat');
+
+// This is an interface that exists only to explicitly declare the stats we are
+// tracking and avoid bugs due to mis-spelled names and types.
+
+function StatEmitter() {
+    var self = this;
+    EventEmitter.call(self);
+    self.statEvent = self.defineEvent('stat');
+}
+
+inherits(StatEmitter, EventEmitter);
+
+// These methods are intended to be used in the constructor to declare stats.
+StatEmitter.prototype.defineCounter = function defineCounter(name) {
+    var self = this;
+    return new Counter(name, self);
+};
+
+StatEmitter.prototype.defineGauge = function defineGauge(name) {
+    var self = this;
+    return new Gauge(name, self);
+};
+
+StatEmitter.prototype.defineTiming = function defineTiming(name) {
+    var self = this;
+    return new Timing(name, self);
+};
+
+// This is intended to be overridden for types have piggy-back additional tags.
```

Comment:
if tags are all you need, then I see this level of indirection of over-design:
- just add a tags array on Stat
- support decorating in StatEmitter#emitStat before calling the statEvent.emit

Side benefit, one less touch point on StatEmitter, since "creation" is a static `new Stat`.